### ISSUE TITLE
MD013 line length suddenly started failing

### ISSUE THREAD

**Issue by purpleidea** [markdownlint/markdownlint#576]
## Description
All our builds suddenly started breaking with MD013 (line length)

### Environment
github CI

### Expected Behavior
Things should be passing.

### Actual Behavior
https://github.com/purpleidea/mgmt/actions/runs/26677328201/job/78666828559

For example:

running ./test/test-markdownlint.sh 
README.md:5: MD013 Line length
README.md:6: MD013 Line length
README.md:96: MD013 Line length
README.md:144: MD013 Line length
README.md:151: MD013 Line length
README.md:152: MD013 Line length
README.md:155: MD013 Line length

The README is here: https://github.com/purpleidea/mgmt/

## Replication Case
Full test case we use is right here: https://github.com/purpleidea/mgmt/blob/master/test/test-markdownlint.sh

Thanks!

**Comment by jaymzh** [comment]
The rule was fixed in a variety of ways. It both used to give false positives (on lines with only URLs in the case it was indented for a code block or UL) and false negatives (in cases like yours).

Now, it's as-described.

Your line, is indeed, too long. It should broken where appropriate, such as (to take the first example):

```
[![Go Report Card](
https://goreportcard.com/badge/github.com/purpleidea/mgmt?style=flat-square
)](https://goreportcard.com/report/github.com/purpleidea/mgmt)
```

**Comment by purpleidea** [comment]
Oh hey Phil. Long time!

Going through the breakages and there are gems like this:

> [debugging golang programs](https://purpleidea.com/blog/2016/02/15/debugging-golang-programs/)

It wants me to split this over multiple lines. In no way do I think that's an improvement in the many types of places this happens. I don't suppose there is an option to "leave single URL's alone" ? Yes I realize I could split the url text, but this really does not improve readability or the spirit of the line length rule in any way from my POV.

If there's not such an option, would you accept a new filter eg MD99X or something that restores the previous behaviour for URL's so we could ignore MD013 but add MD99X instead?

**Comment by jaymzh** [comment]
Heya!

No need to add a new rule. Feel free to add a new option to the existing rule though!

**Comment by jaymzh** [comment]
@purpleidea - `treat_links_as_single_words` is now added, plus we fixed a bunch of other false-positive cases. 0.17.0 is out.

**Comment by purpleidea** [comment]
@jaymzh Sweet!

This is certainly an improvement, thanks! Here's a related situation to that:

```
[debugging golang programs](https://purpleidea.com/blog/2016/02/15/debugging-golang-programs/).
```

The period following this sentence puts it over the edge. So if sentences end with a link (minus the punctuation) the period would need to go on a line by itself.

Which I still think is weird.

I will go through the source now and find you a list of such test cases if that would help.

**Comment by jaymzh** [comment]
Yeah that's not a case I want to support... The special casing will get out of hand. MD renderers already collapse newlines. Fwiw, i do that like:

```
Here is a [link](
URL
).
```
